examples/external-data/tests/customerService.test.ts

```diff
@@ -154,6 +154,14 @@ describe("mock-customer-service", () => {
 
 		await closeServer(_externalDataService);
 		await closeServer(_customerService);
+
+		// Something about shutting down the servers after each test and then starting new ones on the same ports before
+		// running the next test is causing issues where the second test to run gets an "other side closed" message when
```

Comment:
server.close doesn't close connections that are mid-request, I wonder if the callback gets called anyway.

server.closeAllConnections() is more aggressive and might be an option?  Or I wonder if waiting for the server to emit the "close" event would work instead of relying on the callback?

Reply:
Couldn't make it work with either if there was no `setTimeout` here :(